### DISCUSSION TITLE
EntityManager bean only for build time

### DISCUSSION THREAD

**Discussion by mgafarov** [quarkusio/quarkus#42025] [Q&A]
I want to split my monolith to modules,1 of them as library with some common hibernate classes where EntityManager is injected.

But I don't want this module to initialize database connection or EntityManager and provide all of them from other modules where this module will be added as library. But if they are not initialized,then I fail with unsatisfied dependency for EntityManager at build time.

Can I somehow disable EntityManager lookup as bean for injection or create some dummy EntityManager only for build time? Or is there any other solution?

**Comment by hrgdavor** [comment]
long time since asked, and no answers. I am looking into using quarkus to avoid spring-boot, but this is concerning.

**Reply by gsmet** [reply]
> long time since asked, and no answers. I am looking into using quarkus to avoid spring-boot, but this is concerning.

Not sure what concerns you. The fact that it doesn't have any answers (it happens, we sometimes miss some questions) or the issue itself. If it's the issue itself, see the answer I just posted below!

**Reply by hrgdavor** [reply]
I just started exploring the repo, so lack of answer was concerning from a perspective of someone that does not know how active quarkus is. It is nice to see that  there was no need for concern :)

**Reply by gsmet** [reply]
Yeah, I wouldn’t worry too much about that. The project is extremely active. This why sometimes things call through the cracks :)

**Comment by gsmet** [comment]
If you don't need Quarkus itself in a library module, just depend on the JPA API, for instance. Or the `hibernate-core` artifact if you need some Hibernate ORM-specific features.
You don't need the Quarkus runtime in a library module.

Just make sure the library module is properly indexed with Jandex. See my answer on StackOverflow for the details: https://stackoverflow.com/a/55513723/5043585 .
